client/src/modules/stock/entry/entry.js

```diff
@@ -537,6 +552,14 @@ function StockEntryController(
 
     movement.lots = Stock.processLotsFromStore(vm.stockForm.store.data, vm.movement.entity.uuid);
 
+    // fix the unit price, if necessary
+    if (vm.currencyId !== vm.enterprise.currency_id) {
+      const exchangeRate = Exchange.getCurrentRate(vm.currencyId);
+      movement.lots.forEach(lot => {
+        lot.unit_cost /= exchangeRate;
```

Comment:
This is good ... until an error occurs.  Not sure why an error would occur, but that would cause this code to doubly exchange.

We should reset the `unit_cost` if an error occurs.  I can make this change.

Reply:
Were you able to fix this or should this be noted as an Issue?